api/app/controllers/spree/api/base_controller.rb

```diff
@@ -133,13 +133,13 @@ def find_product(id)
 
       def product_scope
         if can?(:admin, Spree::Product)
-          scope = Spree::Product.with_deleted.accessible_by(current_ability, :read).includes(*product_includes)
+          scope = Spree::Product.with_deleted.accessible_by(current_ability).includes(*product_includes)
```

Comment:
this is not necessary anymore. The default is `:index` which is by default aliased by `:read`.

Reply:
Hi @coorasse! But if `:read` alias was `:show` shouldn't it be changed like this `scope = Spree::Product.with_deleted.accessible_by(current_ability, :show).includes(*product_includes)` to keep same logic? Isn't your change changing scope from show to index now?

Reply:
it changes from `:show` to `:index` but all the rules are defined for `:read` instead of `:display` which was including also `:read`. it's quite complicated, I know, I don't know why all these aliases were generated before.  `accessible_by(ability, :show)` would not make sense because you want a list of objects. `:show` is for single instances.

Reply:
Yeah, that make sense. So for now, when is `alias_action :show, to: :read` this scope with argument `(current_ability, :read)` had no authorization at all I gues?

Reply:
`:read` was aliased by `:display` and rules were (almost) always defined on `:display`. You can see that most of the changes are about that: https://github.com/solidusio/solidus/pull/3148/files#diff-82b30f937458e7c1052c326354dc18cbL8

Reply:
I agree that `:index` is the right thing here, as `:show` would need to check every single instance of a Product if its able to be shown to the user, what is definitely not what we want here.